apache/superset
# feat(db_engine): Implement user impersonation support for StarRocks

### SUMMARY

This PR adds user impersonation support for the StarRocks database by leveraging the native `EXECUTE AS {user} WITH NO REVERT` user management query.

### TESTING INSTRUCTIONS

1. If necessary create a separate user on your StarRocks DB and your Superset instance. The names must match and it should be different from the user that is used for the StarRocks connection itself
2. Enable the user impersonation feature in the security settings of your StarRocks database
3. Log in to Superset with your testing user
4. Execute `SELECT CURRENT_USER();` and see that it shows your current Superset user
5. Disable user impersonation feature again
6. Execute `SELECT CURRENT_USER();` again and see that it changed back to the user of the StarRocks connection

### ADDITIONAL INFORMATION
